### PR TITLE
Add closeDatabase fn to gracefully close DB connections

### PR DESCRIPTION
**Pull request by tanuj-shardeum**

### CODE REVIEW

**Comment by justin-shardeum** [comment]
fixes BLUE-55

File: src/distributor.ts [hunk 1]
```diff
@@ -72,9 +73,23 @@ const addSigListeners = (): void => {
       // Refresh the list of subscribers in every worker process
       updateConfigAndSubscriberList()
     }
+    await dbstore.closeDatabase()
+    process.exit(0)
```
Comment (jairajdev) [inline src/distributor.ts]: Actually, we don't exit when SIGUSR1 event is triggered. it's to refresh the config changes

File: src/distributor.ts [hunk 2]
```diff
@@ -72,9 +73,23 @@ const addSigListeners = (): void => {
...
+  })
+  process.on('SIGTERM', async () => {
+    console.log('Exiting on SIGTERM')
+    await dbstore.closeDatabase()
+    process.exit(0)
+  })
+  process.on('uncaughtException', async (error) => {
     console.error('Uncaught Exception in Distributor: ', error)
+    await dbstore.closeDatabase()
+    process.exit(0)
```
Comment (jairajdev) [inline src/distributor.ts]: I think we shouldn't exit on `uncaughtException` exception too.

**Comment by jairajdev** [comment]
@tanuj-shardeum I added some feedbacks. thanks.

**Review by jairajdev — APPROVED** [review]

**Pull request merged**